### PR TITLE
ast, plan: return error when the arg of VALUES() function is not a column (#7817)

### PR DESCRIPTION
**Pull request by XuHuaiyu**
cherry-pick #7817

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8404)

### CODE REVIEW

**Comment by XuHuaiyu** [comment]
/run-all-test tidb-test=release-2.0 pd=release-2.0 tikv=release-2.0

File: plan/logical_plan_test.go [hunk 1]
```diff
@@ -1644,6 +1644,8 @@ func (s *testPlanSuite) TestNameResolver(c *C) {
 		{"select a from t group by t11.c1", "[planner:1054]Unknown column 't11.c1' in 'group statement'"},
 		{"delete a from (select * from t ) as a, t", "[planner:1288]The target table a of the DELETE is not updatable"},
 		{"delete b from (select * from t ) as a, t", "[planner:1109]Unknown table 'b' in MULTI DELETE"},
+		{"select '' as fakeCol from t group by values(fakeCol)", "[planner:1054]Unknown column '' in 'VALUES() function'"},
+		{"update t, (select * from t) as b set b.a = t.a", "[planner:1288]The target table b of the UPDATE is not updatable"},
```
Comment (eurekaka) [inline plan/logical_plan_test.go]: This line is irrelevant?

Comment (XuHuaiyu) [reply plan/logical_plan_test.go]: Yes, removed now.

**Comment by zz-jason** [comment]
/run-all-tests tidb-test=release-2.0 tikv=release-2.0 pd=release-2.0

**Pull request merged**